src/command_controller.rs

```diff
@@ -0,0 +1,288 @@
+            metadata::object_to_owner_reference::<T::Owner>(owner.meta().clone(), true)?;
+
+        patch_owner_reference(
+            &self.context.client,
+            &self.context.resource,
+            &owner_reference,
+        )
+        .await?;
+
+        Ok(ReconcileFunctionAction::Done)
+    }
+}
+
+impl<T> ReconciliationState for CommandState<T>
+where
+    T: CommandOwner + Clone + DeserializeOwned + Meta + Send + Sync,
+{
+    type Error = Error;
+
+    fn reconcile(
+        &mut self,
+    ) -> Pin<Box<dyn Future<Output = Result<ReconcileFunctionAction, Self::Error>> + Send + '_>>
+    {
+        Box::pin(async move { self.set_owner_reference().await })
+    }
+}
+
+#[derive(Debug)]
+struct CommandStrategy<T> {
+    // TODO: Better workaround for PhantomData?
+    // We use it here cause we need to make CommandStrategy generic to be able to do:
+    // impl<T> ControllerStrategy for CommandStrategy<T>
+    _ignore: Option<std::marker::PhantomData<T>>,
+}
+
+impl<T> CommandStrategy<T> {
+    pub fn new() -> CommandStrategy<T> {
+        CommandStrategy { _ignore: None }
+    }
+}
+
+#[async_trait]
+impl<T> ControllerStrategy for CommandStrategy<T>
+where
+    T: CommandOwner + Meta + Clone + DeserializeOwned + Send + Sync,
+{
+    type Item = T;
+    type State = CommandState<T>;
+    type Error = Error;
+
+    fn finalizer_name(&self) -> String {
+        FINALIZER_NAME.to_string()
+    }
+
+    async fn init_reconcile_state(
+        &self,
+        context: ReconciliationContext<Self::Item>,
+    ) -> Result<Self::State, Error> {
+        Ok(CommandState { context })
+    }
+}
+
+/// This creates an instance of a [`Controller`] which waits for incoming events and reconciles them.
+///
+/// This is an async method and the returned future needs to be consumed to make progress.
+pub async fn create_command_controller<T>(client: Client)
+where
+    T: CommandOwner + Meta + Clone + Debug + DeserializeOwned + Send + Sync + 'static,
+{
+    let command_api: Api<T> = client.get_all_api();
+    let owner_api: Api<T::Owner> = client.get_all_api();
+
+    let controller = Controller::new(command_api).owns(owner_api, ListParams::default());
+
+    let strategy = CommandStrategy::new();
+
+    controller
+        .run(client, strategy, Duration::from_secs(10))
+        .await;
+}
+
+/// Find a resource according to the metadata.name field.
+///
+/// # Arguments
+/// * `client` - Kubernetes client
+/// * `metadata_name` - The metadata.name value to be matched
+///
+async fn find_owner<T>(client: &Client, metadata_name: &str) -> OperatorResult<T>
+where
+    T: Clone + DeserializeOwned + Meta,
+{
+    let lp = ListParams::default().fields(&format!("metadata.name={}", metadata_name));
+    let mut owners: Vec<T> = client.list(None, &lp).await?;
+
+    if owners.is_empty() {
+        return Err(Error::MissingCustomResource {
+            name: metadata_name.to_string(),
+        });
+    }
+
+    Ok(owners.pop().unwrap())
```

Comment:
You could change this (including the is_empty) to a match on pop() (but I'm fine with the way it is now)